sound/soc/sof/pm.c

```diff
@@ -27,16 +27,16 @@ static int sof_restore_kcontrols(struct snd_sof_dev *sdev)
 		case SOF_CTRL_CMD_VOLUME:
 			ipc_cmd = SOF_IPC_COMP_SET_VALUE;
 			ctrl_type = SOF_CTRL_TYPE_VALUE_CHAN_SET;
-			ret = snd_sof_ipc_set_comp_data(sdev->ipc, scontrol,
-							ipc_cmd, ctrl_type,
-							scontrol->cmd);
+			ret = snd_sof_ipc_set_get_comp_data(sdev->ipc, scontrol,
+							    ipc_cmd, ctrl_type,
+							    scontrol->cmd);
 			break;
 		case SOF_CTRL_CMD_BINARY:
 			ipc_cmd = SOF_IPC_COMP_SET_DATA;
 			ctrl_type = SOF_CTRL_TYPE_DATA_SET;
-			ret = snd_sof_ipc_set_comp_data(sdev->ipc, scontrol,
-							ipc_cmd, ctrl_type,
-							scontrol->cmd);
+			ret = snd_sof_ipc_set_get_comp_data(sdev->ipc, scontrol,
```

Comment:
@juimonen I'm wondering if we should do this right here or maybe schedule it as a work to be done in the future? Depending on the size of the data this could take a really long time and it will affect the resume latency. For example, on the GLK we had a 1s resume latency requirement.